javalib/src/main/scala/java/util/Collections.scala

```diff
@@ -0,0 +1,1131 @@
+    sort(list, makeComparator[T])
+
+  def sort[T](list: List[T], c: Comparator[_ >: T]): Unit = {
+    val sortedList = list.sorted(c)
+    (0 until list.size).foreach(i => list(i) = sortedList(i))
+  }
+
+  def binarySearch[T](list: List[_ <: jl.Comparable[T]], key: T): Int =
+    binarySearchCore(list, (elem: Comparable[T]) => elem.compareTo(key))
+
+  def binarySearch[T](list: List[_ <: T], key: T, c: Comparator[_ >: T]): Int =
+    binarySearchCore(list, (elem: T) => c.compare(elem, key))
+
+  def reverse(list: List[_]): Unit = {
+    val _list = list.asInstanceOf[List[AnyRef]]
+    val it1 = _list.listIterator()
+    val it2 = _list.listIterator(_list.size)
+    for (i <- 0 until _list.size / 2) {
+      val tmp = it1.next()
+      it1.set(it2.previous())
+      it2.set(tmp)
+    }
+  }
+
+  def shuffle(list: List[_]): Unit =
+    shuffle(list, new Random)
+
+  def shuffle(list: List[_], rnd: Random): Unit = {
+    val _list = list.asInstanceOf[List[AnyRef]]
+    val scalaRnd = scala.util.Random.javaRandomToRandom(rnd)
+    val shuffledList = scalaRnd.shuffle(_list.toSeq)
+    (0 until list.size).foreach(i => _list(i) = shuffledList(i))
+  }
+
+  def swap(list: List[_], i: Int, j: Int): Unit =
+    swapCore(list, i, j, true)
+
+  def fill[T](list: List[_ >: T], obj: T): Unit = {
+    @tailrec
+    def fillNext(iter: ListIterator[_ >: T]): Unit = {
+      if (iter.hasNext) {
+        iter.next()
+        iter.set(obj)
+        fillNext(iter)
+      }
+    }
+    fillNext(list.listIterator)
+  }
+
+  def copy[T](dest: List[_ >: T], src: List[_ <: T]): Unit = {
+    @tailrec
+    def copy(iterDest: ListIterator[_ >: T], iterSrc: ListIterator[_ <: T]): Unit = {
+      if (iterSrc.hasNext) {
+        if (iterDest.hasNext) {
+          iterDest.next()
+          iterDest.set(iterSrc.next())
+          copy(iterDest, iterSrc)
+        } else
+          throw new IndexOutOfBoundsException
+      }
+    }
+    copy(dest.listIterator, src.listIterator)
+  }
+
+  def min[T <: AnyRef with jl.Comparable[T]](coll: Collection[_ <: T]): T =
+    min(coll, makeComparator[T])
+
+  def min[T](coll: Collection[_ <: T], comp: Comparator[_ >: T]): T =
+    coll.min(comp)
+
+  def max[T <: AnyRef with jl.Comparable[T]](coll: Collection[_ <: T]): T =
+    max(coll, makeComparator[T])
+
+  def max[T](coll: Collection[_ <: T], comp: Comparator[_ >: T]): T =
+    coll.max(comp)
+
+  def rotate(list: List[_], distance: Int): Unit = {
+    val listSize = list.size
+    if (listSize > 1 && distance % listSize != 0) {
+      def exchangeRotation(): Unit = {
+        val _list = list.asInstanceOf[List[AnyRef]]
+        @tailrec
+        def rotateNext(index: Int, value: AnyRef): Unit = {
+          val nextValue = _list(index)
+          _list(index) = value
+          if (index != 0)
+            rotateNext(modulo(index + distance, listSize), nextValue)
+        }
+        rotateNext(modulo(distance, listSize), _list(0))
```

Comment:
I'm pretty sure this is buggy for values of `distance` that are not prime wrt `list.size`. For example, it seems obvious that for `list.size = 6` and `distance = 2`, only the elements at indices 0, 2 and 4 will be displaced, but elements at indices 1, 3 and 5 will stay put.